plugins/api/README.md

```diff
@@ -111,6 +115,36 @@ data class ExampleConfiguration(
 
 Here, the `serverUrl` property has a default value, the `timeout` property is required, and the `token` property is optional.
 
+The generated `ExampleAdvisorFactory` class will contain two factory functions:
+
+```kotlin
+override fun create(config: PluginConfig): ExampleAdvisor {
+  val configObject = ExampleConfiguration(
+      serverUrl = config.options["serverUrl"] ?: "https://example.com",
+      timeout = config.options["timeout"]?.toInt(),
+      token = config.secrets["token"]
+  )
+
+  return ExampleAdvisor(descriptor, configObject)
+}
+
+companion object {
+  fun create(
+    serverUrl: String = "https://example.com",
+    timeout: Int,
```

Comment:
Looks like the types don't match as in line 124 `timeout` could be `null`, or?

Reply:
The `error("...")` fallback was missing.